django/utils/dateformat.py

```diff
@@ -259,16 +260,14 @@ def o(self):
 
     def r(self):
         "RFC 5322 formatted date; e.g. 'Thu, 21 Dec 2000 16:01:07 +0200'"
-        if type(self.data) is datetime.date:
-            raise TypeError(
-                "The format for date objects may not contain time-related "
-                "format specifiers (found 'r')."
-            )
```

Comment:
This is an unnecessary restriction. If we have a `datetime.date` object we should just assume midnight UTC.